airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/ConfigsDatabaseInstance.java

```diff
@@ -23,19 +25,64 @@
   private static final Function<Database, Boolean> IS_CONFIGS_DATABASE_READY = database -> {
     try {
       LOGGER.info("Testing if airbyte_configs has been created and seeded...");
-      return database.query(ctx -> hasData(ctx, "airbyte_configs"));
+      if (database.query(ctx -> hasTable(ctx, "airbyte_configs")) && database.query(ctx -> hasData(ctx, "airbyte_configs"))) {
+        return true;
+      }
+
+      return database.query(ctx -> hasTable(ctx, "state"));
     } catch (Exception e) {
       return false;
     }
   };
 
   @VisibleForTesting
   public ConfigsDatabaseInstance(final String username, final String password, final String connectionString, final String schema) {
-    super(username, password, connectionString, schema, DATABASE_LOGGING_NAME, ConfigsDatabaseSchema.getTableNames(), IS_CONFIGS_DATABASE_READY);
+    super(username, password, connectionString, schema, DATABASE_LOGGING_NAME, ConfigsDatabaseTables.getTableNames(), IS_CONFIGS_DATABASE_READY);
   }
 
   public ConfigsDatabaseInstance(final String username, final String password, final String connectionString) throws IOException {
     this(username, password, connectionString, MoreResources.readResource(SCHEMA_PATH));
   }
 
+  @Override
+  public boolean isInitialized() throws IOException {
+    final Database database = Databases.createPostgresDatabaseWithRetry(
+        username,
+        password,
+        connectionString,
+        isDatabaseConnected(databaseName));
+    return new ExceptionWrappingDatabase(database).transaction(ctx -> {
+      if (hasTable(ctx, "state") || hasTable(ctx, "airbyte_configs")) {
```

Comment:
can we leave a quick comment to explain why we specifically check for the 'state' table. thanks